wire-ios-sync-engine/Source/Synchronization/Strategies/EvaluateOneOnOneConversationsStrategy.swift

```diff
@@ -0,0 +1,91 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireRequestStrategy
+
+final class EvaluateOneOnOneConversationsStrategy: AbstractRequestStrategy {
+
+    let syncPhase: SyncPhase = .evaluate1on1ConversationsForMLS
+
+    private unowned var syncProgress: SyncProgress
+
+    private var isSyncing: Bool { syncProgress.currentSyncPhase == syncPhase }
+
+    private var task: Task<Void, Never>?
+
+    public init(
+        withManagedObjectContext managedObjectContext: NSManagedObjectContext,
+        applicationStatus: ApplicationStatus,
+        syncProgress: SyncProgress
+    ) {
+        self.syncProgress = syncProgress
+
+        super.init(withManagedObjectContext: managedObjectContext, applicationStatus: applicationStatus)
+    }
+
+    deinit {
+        task?.cancel()
+    }
+
+    override func nextRequest(for apiVersion: APIVersion) -> ZMTransportRequest? {
+        guard isSyncing, task == nil else {
+            return nil
+        }
+
+        WireLogger.conversation.info("EvaluateOneOnOneConversationsStrategy: start evaluate one on one conversations!")
+
+        precondition(managedObjectContext.zm_isSyncContext, "can only execute on syncContext!")
+        let syncContext = managedObjectContext
+
+        // store task to avoid duplicated entries and concurrency issues
+        task = Task { [weak self] in
+            guard let self else {
+                assertionFailure("EvaluateOneOnOneConversationsStrategy: cannot perform without self reference!")
+                return
+            }
+
+            do {
+                let resolver = OneOnOneResolver(syncContext: syncContext)
+                try await resolver.resolveAllOneOnOneConversations(in: syncContext)
+
+                await syncContext.perform {
+                    self.finishCurrentSyncPhase()
+                }
+            } catch {
+                await syncContext.perform {
+                    self.failCurrentSyncPhase(errorMessage: "EvaluateOneOnOneConversationsStrategy: failed to resolve all 1-1 conversations!")
+                }
+            }
+
+            self.task = nil
+        }
+
+        return nil
+    }
```

Comment:
This isn't making any requests, I guess it's some kind of workaround to make it a slow sync step?

Reply:
Yes it's necessary to fit into the operation loop as it works now to perform the slow sync.
John and me thought about a refactoring, but it would mean also risks of breaking something at this core functionality, that I didn't want to take in the current period before release.